var/spack/repos/builtin/packages/nfft/package.py

```diff
@@ -24,46 +24,46 @@ def configure(self, spec, prefix):
 
         configure = Executable('../configure')
 
-        if '+double' in spec['fftw']:
+        if 'double' in spec['fftw'].variants.get('precision'):
             with working_dir('double', create=True):
                 configure(*options)
-        if '+float' in spec['fftw']:
+        if 'float' in spec['fftw'].variants.get('precision'):
             with working_dir('float', create=True):
                 configure('--enable-float', *options)
-        if '+long_double' in spec['fftw']:
+        if 'long_double' in spec['fftw'].variants.get('precision'):
```

Comment:
Any reason for not leveraging `fftw`'s `selected_precisions`?  Also, you could get the options once at around line 26 then check against the local variable so you don't keep making calls to it.

Same question/suggestinon for `build`, `check`, and `install` below.

Reply:
@tldahlgren To avoid repeated calls to `spec['fftw']` in `configure`, `build`, `check` and `install` we could also save `fftw`'s `selected_precisions` in a class (instance) variable `self.fftw_selected_precisions` that we initialise during `configure`. Would that be beneficial?

Reply:
A second question: Where/how to I access `fftw`'s `selected_precisions`? `spec['fftw']` (type `<class llnl.util.lang.Spec>`) has no attribute `selected_precisions`.

Reply:
> @tldahlgren To avoid repeated calls to `spec['fftw']` in `configure`, `build`, `check` and `install` we could also save `fftw`'s `selected_precisions` in a class (instance) variable `self.fftw_selected_precisions` that we initialise during `configure`. Would that be beneficial?

Adding a property initialized in that manner is certainly an option, e.g.,:

```
    _fftw_precisions = None

    @property
    def fftw_selected_precisions(self):
        if not self._fftw_precisions:
            self._fftw_precisions = self.spec['fftw'].package.selected_precisions
        return self._fftw_precisions
```

Reply:
> A second question: Where/how to I access `fftw`'s `selected_precisions`? `spec['fftw']` (type `<class llnl.util.lang.Spec>`) has no attribute `selected_precisions`.

The example I gave above shows that the `spec`'s `package` has the attribute.

Reply:
Thanks for your help @tldahlgren . I have updated the code following your suggestions.